pnet-idp-client-saml2/src/main/java/at/porscheinformatik/idp/saml2/Saml2Utils.java

```diff
@@ -108,7 +114,11 @@ public static UriComponentsBuilder setRelayState(UriComponentsBuilder uriCompone
 
     public static Optional<String> getRelayState(HttpServletRequest request)
     {
-        return Optional.ofNullable(request.getParameter(RELAY_STATE_PARAM));
+        return Optional.ofNullable(request.getParameter(RELAY_STATE_PARAM))
+            .map(rs -> {
+                Matcher matcher = AUTO_GENERATED_RELAY_STATE_PATTERN.matcher(rs);
+                return matcher.matches() ? matcher.group() : rs;
```

Comment:
Please use `matcher.group(1)` here. `group()` is equivalent to `group(0)`. And group(0) is the whole string. But we only need the user provided part.

Reply:
Sure, changed to `group(1)`